blueflood-http/src/main/java/com/rackspacecloud/blueflood/outputs/handlers/HttpOptionsHandler.java

```diff
@@ -0,0 +1,42 @@
+package com.rackspacecloud.blueflood.outputs.handlers;
+
+import com.rackspacecloud.blueflood.http.DefaultHandler;
+import com.rackspacecloud.blueflood.http.HttpRequestHandler;
+import com.rackspacecloud.blueflood.service.Configuration;
+import com.rackspacecloud.blueflood.service.CoreConfig;
+import com.rackspacecloud.blueflood.tracker.Tracker;
+import org.jboss.netty.channel.ChannelHandlerContext;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * class to handle Cross-Origin Resource Sharing (CORS) OPTIONS requests
+ */
+public class HttpOptionsHandler implements HttpRequestHandler {
+
+    private final boolean CORS_ENABLED = Configuration.getInstance().getBooleanProperty(CoreConfig.CORS_ENABLED);
+    private final String CORS_ALLOWED_ORIGINS = Configuration.getInstance().getStringProperty(CoreConfig.CORS_ALLOWED_ORIGINS);
+    private final String CORS_ALLOWED_METHODS = Configuration.getInstance().getStringProperty(CoreConfig.CORS_ALLOWED_METHODS);
+    private final String CORS_ALLOWED_HEADERS = Configuration.getInstance().getStringProperty(CoreConfig.CORS_ALLOWED_HEADERS);
+    private final String CORS_ALLOWED_MAX_AGE = Configuration.getInstance().getStringProperty(CoreConfig.CORS_ALLOWED_MAX_AGE);
+
+    @Override
+    public void handle(ChannelHandlerContext ctx, HttpRequest request) {
+        // log the request if tracking enabled
+        Tracker.getInstance().track(request);
+
+        // set CORS headers in the response
+        Map<String, String> headers = new HashMap<String, String>();
+        if (CORS_ENABLED) {
+            headers.put("Access-Control-Allow-Origin", CORS_ALLOWED_ORIGINS);
```

Comment:
@VinnyQ Never mind my previous comment. It will make it complicated and we can postpone that feature. But, here we are missing the essence of CORS. eg if someone sets ```CORS_ALLOWED_ORIGINS``` property to ```http://*.rackspace.com``` With the above code, if a request comes from anywhere, we are returning ```Access-Control-Allow-Origin: http://*.rackspace.com``` as response to every single request. 
There should be a method here that should essentially validate the requester's origin and see if it matches with the ```CORS_ALLOWED_ORIGINS``` property.
eg if a request comes from ```http://foo.rackspace.com``` and the ```CORS_ALLOWED_ORIGINS``` property is set to ```http://*.rackspace.com``` then the method should return ```Access-Control-Allow-Origin: http://foo.rackspace.com``` as response headers, along with other headers. Same is the case for ```http://bar.rackspace.com```
This is how you can get the the request origin ```String host = request.getRemoteHost(); ```
Do the pattern matching on that and send the response.
.
.
.
Now this sounds like a lot of work, isn't it and will involve a lot of testing for all the cases.
How about we leverage the features provided by netty server. Unfortunately we are running an older version of netty. Probably 3.x. If we upgrade our server to Version 4.10 That comes with CORS support and you won't have to do all this manual work :blush: